app/src/main/java/org/dhis2/utils/optionset/OptionSetContracts.kt

```diff
@@ -0,0 +1,28 @@
+package org.dhis2.utils.optionset
+
+import android.widget.EditText
+import android.widget.TextView
+import androidx.lifecycle.LiveData
+import androidx.paging.PagedList
+import org.dhis2.data.forms.dataentry.fields.spinner.SpinnerViewModel
+import org.dhis2.usescases.general.AbstractActivityContracts
+import org.hisp.dhis.android.core.option.Option
+
+class OptionSetContracts {
+
+    interface View {
+
+        fun setLiveData(data: LiveData<PagedList<Option>>?)
+
+        fun showDialog(): Boolean
+    }
+
+    interface Presenter : AbstractActivityContracts.Presenter {
+
+        fun init(view: View, optionSet : SpinnerViewModel, textSearch : EditText)
+
+        fun init(view: View, optionSetTable: org.dhis2.data.forms.dataentry.tablefields.spinner.SpinnerViewModel, textSearch: EditText)
```

Comment:
In kotlin we can use importing aliases. So basically we can import SpinnerViewModel from both places as follows.

    import org.dhis2.data.forms.dataentry.fields.spinner.SpinnerViewModel
    import org.dhis2.data.forms.dataentry.tablefields.spinner.SpinnerViewModel **as TableSpinnerViewModel**

And now we can update the table init as follows
    `fun init(view: View, optionSetTable: **TableSpinnerViewModel**, textSearch: EditText)`